File: python/bubble_sort.py
sequence = [4, 3, 5, 0, 1]
swaps = 0
operations = 0

n = len(sequence)
for i in range(n - 1):
    for j in range(0,n - i - 1):
        operations += 1
        # print(f"operations {operations}")
        if  sequence[j] > sequence[j + 1]:
            swaps += 1
            sequence[j],sequence[j + 1] = sequence[j + 1],sequence[j]
            # print(f"sequence[j]: {sequence[j]} , sequence[j + 1]  {sequence[j + 1]}, sequence {sequence}")
            
        

print(f"Final result: {sequence}")
print(f"Swaps: {swaps}")

File: js/bubbleSort.js
// var sequence = [4, 3, 5, 0, 1]
var sequence = [5,4,3,2,1]
var swaps = 0
let iterations = 0


const n = sequence.length

for(let i = 0; i < n; i++) {
    for(let j = 0; j < n - i - 1; j++) {
        iterations ++
        if (sequence[j] > sequence[j + 1]) {
            let temp = sequence[j]
            sequence[j] = sequence[j + 1]
            sequence[j + 1] = temp
            swaps++
            console.log(`sequence[i]: ${sequence[i]} , sequence[i + 1]  ${sequence[i + 1]}, sequence: ${sequence}`)        
        }
    }
} 


// let sorted = false
// while (!sorted) {
//     sorted = true
    
//     for(let i = 0; i < n - 1; i++) {
//         iterations++
//         if(i + 1 === n.lenth) {
//             break
//         }
//         if(sequence[i] > sequence[i + 1]) {
//             swaps++
//             let temp = sequence[i]
//             sequence[i] = sequence[i + 1]
//             sequence[i + 1] = temp
//             sorted = false
//             // break;
//         }
//     }
// }

console.log("Final result: " + sequence)
console.log("Swaps: " + swaps)
console.log("Iterations: " + iterations)